raysect/core/math/quaternion.pyx

```diff
@@ -0,0 +1,556 @@
+
+import numbers
+cimport cython
+from libc.math cimport sqrt, sin, cos, M_PI
+
+from raysect.core.math.affinematrix cimport new_affinematrix3d
+
+
+cdef double DEG2RAD = 2*M_PI / 360
+cdef double RAD2DEG = 360 / 2*M_PI
+
+
+cdef class Quaternion:
+
+    def __init__(self, double s=1.0, double x=0.0, double y=0.0, double z=0.0):
+
+        self.s = s
+        self.x = x
+        self.y = y
+        self.z = z
+
+    def __repr__(self):
+        """Returns a string representation of the Quaternion object."""
+
+        return "Quaternion(" + str(self.s) + ", <" + str(self.x) + ", " + str(self.y) + ", " + str(self.z) + ">)"
```

Comment:
Violates intention of repr. It should either generate a string that can be copy-pasted to reinstance the object or be an alternate, non-code representation enclosed in <blah>. This should return: Quarternion(1, 0, 0, 1)  and not Quarternion(1, <0, 0, 1>) as this is invalid python.

Reply:
This will mean fixing the docstring examples too.

Reply:
Resolved in a4d9d8a